### ISSUE TITLE
use `report_card_template` in `tm_missing_data`

### ISSUE THREAD

**Issue by donyunardi** [insightsengineering/teal.modules.general#634]
https://github.com/insightsengineering/teal.modules.general/blob/2507e36302c9f882be049c1ed3e88645328a353e/R/tm_missing_data.R#L1219

We missed this during this PR:
https://github.com/insightsengineering/teal.modules.general/pull/584

**Comment by m7pr** [comment]
I don't think we did miss `tm_missing_data` as it is changed in this PR https://github.com/insightsengineering/teal.modules.general/pull/584/files#diff-a4da4c6e0574ea76e9c565bcc6c3126f847bb0cefcb1bbb8f5f1dfdeb3cd2e31R1203
You can not use `teal::report_card_template` directly for `tm_missing_data` as in other modules, as it needs to have `label` computed that is based on `sum_type` and `dataname`. Also the `title` is combined from multiple sources.

**Comment by donyunardi** [comment]
ah okay, thanks @m7pr !
